src/drafts/SelectPanel2/SelectPanel.tsx

```diff
@@ -244,7 +254,15 @@ const SelectPanelSearchInput = props => {
 SelectPanel.SearchInput = SelectPanelSearchInput
 
 const SelectPanelFooter = ({...props}) => {
-  const {onCancel} = React.useContext(SelectPanelContext)
+  const {onCancel, selectionVariant} = React.useContext(SelectPanelContext)
+
+  const hidePrimaryActions = selectionVariant === 'instant'
+
+  if (hidePrimaryActions && !props.children) {
+    // nothing to render
+    // todo: we can inform them the developer footer will render nothing
```

Comment:
not throwing validation errors yet. Will create a PR for all validations errors together